contracts/LinniaPermissions.sol

```diff
@@ -59,16 +68,101 @@ contract LinniaPermissions is Ownable, Pausable, Destructible {
         return permissions[dataHash][viewer].canAccess;
     }
 
+    /// Add a delegate for a user's permissions
+    /// @param delegate the address of the delegate being added by user
+    function addDelegate(address delegate)
+    onlyUser
+    whenNotPaused
+    external
+    returns (bool)
+    {
+        delegates[msg.sender][delegate] = true;
```

Comment:
Add the following:
require(delegate != address(0));
require(delegate != msg.sender);
require(delegate != this);